contributing-guides/git-terminal.md

```diff
@@ -29,3 +31,32 @@ Typically a pull request will include changes in a single file.
 [pr-howto]: ../CONTRIBUTING.md#submitting-a-pull-request
 [commit-msg]: ../CONTRIBUTING.md#commit-message
 [mass-changes]: https://github.com/tldr-pages/tldr/pulls?&q=is:pr+is:merged+label:"mass+changes"
+
+# Updating your fork
+
+Forks of GitHub repositories aren't updated automatically. In order to keep your fork of the repository up-to-date with the latest changes and avoid merge conflicts, you should update it regularly.
+
+There are two ways to update your fork.
+
+1. Via the GitHub web interface.  Click `Fetch upstream` and then `Fetch and merge` on the fork as shown below:
+
+![Fetch and merge button in GitHub](../images/github-fetch-and-merge-button.png).
+
+2. Using Git in the terminal:
+
+```bash
+git checkout main
+git remote add upstream https://github.com/tldr-pages/tldr.git  # only run if you don't already have the upstream remote (check with "git remote -v")
+git fetch upstream main
+git rebase upstream/main
+git push --force-with-lease  # not needed if you only want to update your local repository
+```
+
+# Changing the email of your last commit
+
+If the email that you used for a commit isn't added to your GitHub account, you can either add it [here](https://github.com/settings/emails) or change the email of your last commit with the following commands:
```

Comment:
We say "last commit" in the heading, but not switch to "a commit" in the text.
The commands below will only work for the "last" commit.

Also, I think it'd be better to say "email associated" rather than "email added"?

```suggestion
If the email that you used for the last commit isn't associated with your GitHub account, you can either add it [here](https://github.com/settings/emails) or change the email of the commit with the following commands:
```

It might be nice to document how to change the email for a commit that isn't the last one. Could be worth documenting as well?

Reply:
Sounds good, but I don't really understand what `next_comit_to_amend` should be.

Reply:
So imagine the following scenario.

| Git Hash | Author Email
|---|---
| A | sethi@tuta.io
| B | seth@falco.fun
| C | seth@falco.fun
| D | sethi@tuta.io
| E | seth@falco.fun
| F | seth@falco.fun

I need to update `C` and `E` to use my up-to-date email address, the rest are fine.

```bash
git reset A
git commit --amend --author="Seth Falco <seth@falco.fun>"
git cherry-pick B^..D
git commit --amend --author="Seth Falco <seth@falco.fun>"
git cherry-pick E^..HEAD
```

And now A and E both have the correct email address, and we're up-to-date with HEAD again... though kind of in a convoluted manner, to be honest. Much easier than using [git-filter-branch](https://git-scm.com/docs/git-filter-branch), though!

Reply:
Thanks, now I get it. (Amazing how much effort you put into your answers 😄)

I'll add that section and prabably also give an example.